flake.nix

```diff
@@ -0,0 +1,95 @@
+{
+  description =
+    "Decentralized search engine & automatized press reviews - firefox extension";
+
+  # Nixpkgs / NixOS version to use.
+  inputs.nixpkgs.url = "nixpkgs/21.05";
+
+  inputs.meta-press = {
+    url = "https://framagit.org/Siltaar/meta-press-ext";
+    flake = false;
+    submodules = true;
+    type = "git";
+  };
+
+  outputs = { self, nixpkgs, meta-press }:
+    let
+      version = "1.7.7";
+
+      # System types to support.
+      supportedSystems = [ "x86_64-linux" ];
+
+      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
+      forAllSystems = f:
+        nixpkgs.lib.genAttrs supportedSystems (system: f system);
+
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system:
+        import nixpkgs {
+          inherit system;
+          overlays = nixpkgs.lib.attrValues self.overlays;
+        });
+
+    in {
+
+      # A Nixpkgs overlay.
+      overlays = {
+        firefox-meta-press = final: prev: {
+          firefox-meta-press = with final;
```

Comment:
Stylistic thing: `with` is a bit of an [antipattern](https://nix.dev/anti-patterns/language#with-attrset-expression). It might be  better to just inherit the attributes needed from final. That way all of `nixpkgs` isn't brought into scope. For example:
```nix
inherit (final) wrapFirefox firefox-unwrapped fetchFirefoxAddon
...
```